main.tf

```diff
@@ -9,7 +9,7 @@ locals {
   iam_role_enabled       = local.enabled && var.create_iam_role
   account_id             = data.aws_caller_identity.current.account_id
   partition              = data.aws_partition.current.partition
-  security_group_ids     = var.create_security_group ? concat(var.associated_security_group_ids, [module.mwaa_security_group.id]) : var.associated_security_group_ids
+  security_group_ids     = var.create_security_group ? concat(var.allowed_security_group_ids, [module.mwaa_security_group.id]) : var.allowed_security_group_ids
```

Comment:
we need to use both vars:

```
variable "associated_security_group_ids" {
  type        = list(string)
  default     = []
  description = <<-EOT
    A list of IDs of Security Groups to associate the created resource with, in addition to the created security group.
    These security groups will not be modified and, if `create_security_group` is `false`, must have rules providing the desired access.
    EOT
}

##
## allowed_* inputs are optional, because the same thing can be accomplished by
## providing `additional_security_group_rules`. However, if the rules this
## module creates are non-trivial (for example, opening ports based on
## feature settings, see https://github.com/cloudposse/terraform-aws-msk-apache-kafka-cluster/blob/3fe23c402cc420799ae721186812482335f78d24/main.tf#L14-L53 )
## then it makes sense to include these.
## Reasons not to include some or all of these inputs include
## - too hard to implement
## - does not make sense (particularly the IPv6 inputs if the underlying resource does not yet support IPv6)
## - likely to confuse users
## - likely to invite count/for_each issues
variable "allowed_security_group_ids" {
  type        = list(string)
  default     = []
  description = <<-EOT
    A list of IDs of Security Groups to allow access to the security group created by this module.
    The length of this list must be known at "plan" time.
    EOT
}
```

then 
```
security_group_ids     = var.create_security_group ? concat(var.associated_security_group_ids, [module.mwaa_security_group.id]) : var.associated_security_group_ids
```

`allowed_security_group_ids` are external/existing SGs that are allowed to connect to the created SG for MWAA.

`associated_security_group_ids` are additional (existing) SGs that the resources is placed in (together with the created SG if any)